ABLASTR: Pass `GridType` instead of Global

ax3l opened the pull request:
We cannot use global variables from WarpX in the ABLASTR library, otherwise it will not build on its own or in ImpactX.

Follow-up to #4922

ax3l commented on Source/ablastr/utils/Enums.H:
```diff
@@ -0,0 +1,38 @@
+
+namespace ablastr::utils::enums
+{
+    /** Type of grids used in a simulation:
+     *
+     * Collocated at the same location (AMReX: all "NODAL"), staggered (Yee-style), or hybrid.
+     */
+    enum struct GridType {
+        Collocated = 0,
+        Staggered = 1,
+        Hybrid = 2
+    };
+
+    /** Mesh-refinement patch
+     *
+     * The current (fine) or lower-level (coarse) resolution.
+     */
```
@EZoni @RemiLehe @oshapoval pls let me know if these doc strings are reasonable or if you would change something :pray:

RemiLehe commented on Source/ablastr/utils/Enums.H:
```diff
@@ -0,0 +1,34 @@
+
+    /** Mesh-refinement patch
+     *
+     * The current (fine) or lower-level (coarse) resolution.
```
```suggestion
     * The fine or coarse patch (in terms of spatial resolution) on the same MR level.
     * https://warpx.readthedocs.io/en/latest/theory/amr.html
```

EZoni reviewed (commented):
Looks good to me, thanks! Can you just confirm that moving `PatchType` to ABLASTR does not require any additional changes (wherever `PatchType` is used in the code) at this time?

ax3l commented:
Correct, I moved it over because it categorically is general enough for ABLASTR when introducing this new Enums header. I added a `using namespace` in WarpX to avoid any changes at this point caused by it
